File: backend/bookCar/src/main/java/com/freelance/bookCar/dto/response/booking/GetBookingResponse.java
package com.freelance.bookCar.dto.response.booking;

import com.freelance.bookCar.models.product.voucher.Voucher;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class GetBookingResponse {
    private Integer id;
    private LocalDateTime dateBook;
    private double totalPrice;
    private Integer idUser;
    private Integer paymentMethod;
    private Integer invoice;
    private Voucher voucher;
    private String type;
    private Integer idTour;
}

File: frontend/Angular/src/app/models/response/booking/get-booking-response.ts
import { GetVoucherResponse } from "../product/voucher/voucher/get-voucher-response";

export class GetBookingResponse{
    id?: number;
    dateBook?: Date;
    totalPrice?: number;
    idUser?: number;
    voucher?: GetVoucherResponse;
    paymentMethod?: number;
    invoice?: number;
    type?: string;

    constructor(
        id?: number,
        dateBook?: Date,
        totalPrice?: number,
        idUser?: number,
        paymentMethod?: number,
        invoice?: number
        
    ) {
        this.id = id;
        this.dateBook = dateBook;
        this.totalPrice = totalPrice;
        this.idUser = idUser;
        this.paymentMethod = paymentMethod;
        this.invoice = invoice;
    }
}